tests/arp/test_wr_arp.py

```diff
@@ -0,0 +1,173 @@
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        ptfhost.copy(src="scripts/ferret.py", dest="/opt")
+
+        result = duthost.shell(
+            cmd='''ip route show proto zebra type unicast |
+            sed -e '/default/d' -ne '/0\//p' |
+            head -n 1 |
+            sed -ne 's/0\/.*$/1/p'
+            '''
+        )
+        assert len(result['stderr_lines']) == 0, 'Could not obtain DIP'
+
+        dip = result['stdout']
+        logger.info('VxLan Sender {0}'.format(dip))
+
+        ptfhost.host.options['variable_manager'].extra_vars.update({
+            'ferret_args': '-f /tmp/vxlan_decap.json -s {0}'.format(dip)
+        })
+
+        logger.info('Copying ferret config file to {0}'.format(ptfhost.hostname))
+        ptfhost.template(src='ferret.conf.j2', dest='/etc/supervisor/conf.d/ferret.conf')
+
+        logger.info('Generate pem and key files for ssl')
+        ptfhost.command(
+            cmd='''openssl req -new -x509 -keyout test.key -out test.pem -days 365 -nodes 
+            -subj "/C=10/ST=Test/L=Test/O=Test/OU=Test/CN=test.com"''',
+            chdir='/opt'
+        )
+
+        self.__prepareVxlanConfigData(duthost, ptfhost)
+
+        logger.info('Refreshing supervisor control with ferret configuration')
+        ptfhost.shell('supervisorctl reread && supervisorctl update')
+
+    @pytest.fixture(scope='class', autouse=True)
+    def copyPtfDirectory(self, ptfhost):
+        '''
+        Copys PTF directory to PTF host. This class-scope fixture runs once before test start
+
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        ptfhost.copy(src="ptftests", dest="/root")
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setupRouteToPtfhost(self, duthost, ptfhost):
+        '''
+        Sets routes up on DUT to PTF host. This class-scope fixture runs once before test start
+
+        @param duthost: AnsibleHost object of DUT
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        result = duthost.shell(cmd="ip route show table default | sed -n 's/default //p'")
+        assert len(result['stderr_lines']) == 0, 'Could not find the gateway for management port'
+
+        gwIp = result['stdout']
+        ptfIp = ptfhost.host.options['inventory_manager'].get_host(ptfhost.hostname).vars['ansible_host']
+
+        route = duthost.shell(cmd='ip route get {0}'.format(ptfIp))['stdout']
+        if 'PortChannel' in route:
+            logger.info("Install explicit route through eth0 (mgmt) interface, when we don't have correct route to ptf")
+            duthost.shell(cmd='ip route add {0}/32 {1}'.format(ptfIp, gwIp))
+
+    @pytest.fixture(scope='class', autouse=True)
+    def removePtfhostIp(self, ptfhost):
+        '''
+        Removes IP assigned to eth<n> inerface of PTF host. This class-scope fixture runs once before test start
+
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        ptfhost.script(src='scripts/remove_ip.sh')
```

Comment:
Does this path depend on what directory you run the test from?

Reply:
Yes

Reply:
Could you base it off the location relative to `test_wr_arp.py`, sort of like I did for `docker.py`?

@yxieca it might also be worth defining a common "root" somewhere so everyone doesn't have to do this song and dance every time we want to reference a file in another directory.

Reply:
> Could you base it off the location relative to `test_wr_arp.py`, sort of like I did for `docker.py`?
> 
> @yxieca it might also be worth defining a common "root" somewhere so everyone doesn't have to do this song and dance every time we want to reference a file in another directory.

NACK, Actually pytest starts search of its root and so it is relative the search path of pytest. That was the case with other test cases

Reply:
Right, so I guess my question is: does that still hold if I run `test_wr_arp.py` in `tests/arp` or `tests`?

I'm less worried about this specific test case and more interested in the general behavior. If pytest can resolve the path regardless of where the file is in the overall directory structure that makes our lives a lot easier. If not we have to be careful when we're re-organizing the directory structure with the community. We should make sure we understand the exact behavior so we know if this is something to watch out for as we're having those discussions. :)

Reply:
Pytest-ansible integrates with Ansible framework and root dir is resolved based on where ansible.cfg is. In this case it is under tests. If one wanted to run test cases from a different dir, you would lose all defs in the ansible.cfg and ssh'ing into hosts may even not be possible! Let alone the hardcoding of path in the tests/conftest.py where the topo file is resolved this way
`with open("../ansible/vars/topo_{}.yml".format(topo), 'r') as fh:`

In brief, it may not be possible to run a test case by changing into the test case respective dir. So, I am wondering if we are going to make it a requirement to run test cases from any dir with the tree?

Reply:
I think based on our discussion earlier it's only important that this behavior is consistent from the `tests` directory, so that we can take advantage of the marks + cli helpers that Neetha is proposing. If we can run this test from the `tests` directory and everything gets resolved correctly then that's fine. We should make sure this behavior is clearly documented somewhere though, otherwise this question will probably come up again.